advanced_source/static_quantization_tutorial.rst

```diff
@@ -0,0 +1,675 @@
+    myModel.eval()  
+
+    # Fuse Conv, bn and relu  
+    myModel.fuse_model()  
+
+    # Specify quantization configuration  
+    # Start with simple min/max range estimation and per-tensor quantization of weights 
+    myModel.qconfig = torch.quantization.default_qconfig  
+    print(myModel.qconfig)  
+    torch.quantization.prepare(myModel, inplace=True) 
+
+    # Calibrate first 
+    print('Post Training Quantization Prepare: Inserting Observers')  
+    print('\n Inverted Residual Block:After observer insertion \n\n', myModel.features[1].conv) 
+
+    # Calibrate with the training set 
+    evaluate(myModel, criterion, data_loader, neval_batches=num_calibration_batches)  
+    print('Post Training Quantization: Calibration done') 
+
+    # Convert to quantized model  
+    torch.quantization.convert(myModel, inplace=True) 
+    print('Post Training Quantization: Convert done') 
+    print('\n Inverted Residual Block: After fusion and quantization, note fused modules: \n\n',myModel.features[1].conv) 
+
+    print("Size of model after quantization") 
+    print_size_of_model(myModel)  
+
+    top1, top5 = evaluate(myModel, criterion, data_loader_test, neval_batches=num_eval_batches) 
+    print('Evaluation accuracy on %d images, %2.2f'%(num_eval_batches * eval_batch_size, top1.avg))
+  
+For this quantized model, we see a significantly lower accuracy of just ~62% on these same 300  
+images. Nevertheless, we did reduce the size of our model down to just under 3.6 MB, almost a 4x  
+decrease. 
+
+In addition, we can significantly improve on the accuracy simply by using a different 
+quantization configuration. We repeat the same exercise with the recommended configuration for  
+quantizing for x86 architectures. This configuration does the following:  
+
+- Quantizes weights on a per-channel basis  
+- Uses a histogram observer that collects a histogram of activations and then picks 
+  quantization parameters in an optimal manner. 
+
+.. code:: python
+
+    per_channel_quantized_model = load_model(saved_model_dir + float_model_file)  
+    per_channel_quantized_model.eval()  
+    per_channel_quantized_model.fuse_model()  
+    per_channel_quantized_model.qconfig = torch.quantization.get_default_qconfig('fbgemm')  
+    print(per_channel_quantized_model.qconfig)  
+
+    torch.quantization.prepare(per_channel_quantized_model, inplace=True) 
+    evaluate(per_channel_quantized_model,criterion, data_loader, num_calibration_batches) 
+    torch.quantization.convert(per_channel_quantized_model, inplace=True) 
+    top1, top5 = evaluate(per_channel_quantized_model, criterion, data_loader_test, neval_batches=num_eval_batches) 
+    print('Evaluation accuracy on %d images, %2.2f'%(num_eval_batches * eval_batch_size, top1.avg)) 
+    torch.jit.save(torch.jit.script(per_channel_quantized_model), saved_model_dir + scripted_quantized_model_file)
+
+  
+Changing just this quantization configuration method resulted in an increase  
```

Comment:
Let us quote the actual accuracies on full imagenet here.

Reply:
is this 67.3% as given in your spreadsheet?

Reply:
Yes, with the default fbgemm qconfig, we see a big improvement due to per-channel for weights and histogram observers

Reply:
What is the baseline for this one?

Reply:
71.9%?

Reply:
Yes, I also added a comment earlier, we need to specify the accuracy of the fp32 model.